juju/juju
# firewaller handles subnet changes

## Description of change

A few key changes so that the firewaller correctly listens to subnet changes in a consuming model, and thus creates the correct ingress rules in a consuming model.

A large chunk of the work is boilerplate - the IngressSubnetsForRelation method is moved from the RemoteRelations facade to the RemoteFirewaller facade.

The remote relation handling of the remote firewaller now uses proper workers to:
- wait for remote relation to be fully registered in both models
- watch relation changes
- watch subnet changes

## QA steps

bootstrap aws
deploy mysql
switch controller
deploy mediawiki
relate mediawiki:db default.mysql
add-space default
add-subnet <172.31.0.0/20> default

The above subnet is found by looking at AWS console. After it is added, use the console to check that the security group on the mysql machine has the subnet added to its ingress rules.
